include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -0,0 +1,1260 @@
+    {
+        mockturtle::stopwatch stop{stats.time_total};
+
+        // first, perform random sampling to find an operational starting point
+        const auto starting_point = find_operational_step_point_via_random_sampling(samples);
+
+        // if no operational point was found within the specified number of samples, return
+        if (!starting_point.has_value())
+        {
+            return op_domain;
+        }
+
+        const auto next_clockwise_point = [](std::vector<step_point>& neighborhood,
+                                             const step_point&        backtrack) noexcept -> step_point
+        {
+            assert(std::find(neighborhood.cbegin(), neighborhood.cend(), backtrack) != neighborhood.cend() &&
+                   "The backtrack point must be part of the neighborhood");
+
+            while (neighborhood.back() != backtrack)
+            {
+                std::rotate(neighborhood.begin(), neighborhood.begin() + 1, neighborhood.end());
+            }
+
+            return neighborhood.front();
+        };
+
+        // find an operational point on the contour starting from the randomly determined starting point
+        const auto contour_starting_point = find_operational_contour_step_point(*starting_point);
+
+        auto current_contour_point = contour_starting_point;
+        auto backtrack_point       = current_contour_point.x == 0 ?
+                                         current_contour_point :
+                                         step_point{current_contour_point.x - 1, current_contour_point.y};
+
+        auto current_neighborhood = moore_neighborhood(current_contour_point);
+        auto next_point           = current_contour_point == backtrack_point ?
+                                        current_neighborhood.front() :
+                                        next_clockwise_point(current_neighborhood, backtrack_point);
+
+        while (next_point != contour_starting_point)
+        {
+            const auto operational_status = is_operational(next_point);
+
+            if (operational_status == operational_domain::operational_status::OPERATIONAL)
+            {
+                backtrack_point       = current_contour_point;
+                current_contour_point = next_point;
+            }
+            else
+            {
+                backtrack_point = next_point;
+            }
+
+            current_neighborhood = moore_neighborhood(current_contour_point);
+            next_point           = next_clockwise_point(current_neighborhood, backtrack_point);
+        }
+
+        log_stats();
+
+        return op_domain;
+    }
+
+  private:
+    /**
+     * The SiDB cell-level layout to investigate.
+     */
+    const Lyt& layout;
+    /**
+     * The specification of the layout.
+     */
+    const TT& truth_table;  // TODO implement the matching of multi-input truth table inputs and BDL pair ordering
```

Comment:
here is a ``TODO``

Reply:
Yes, this is indeed still a TODO for the extension. For now, it can be there as it doesn't show up in the documentation. And I don't want to forget it ;) And the code works without it as is.

Reply:
Perfect!